lib/facility.rb

```diff
@@ -1,14 +1,42 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(data)
+    @name = data[:name]
+    @address = data[:address]
+    @phone = data[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration") 
+      @registered_vehicles << vehicle 
+      vehicle.new_registration_date
+      determine_fee(vehicle)
+      @registered_vehicles
+    else
+      nil
+    end
+  end 
+
+  def determine_fee(vehicle)
+    if vehicle.plate_type == :antique
+      @collected_fees += 25
+    elsif vehicle.plate_type == :ev
+      @collected_fees += 200
+    else
+      @collected_fees += 100
+    end 
+  end
 end
```

Comment:
I really like that you broke this `register_vehicle` method out into a helper method. This slims up your code and helps you stick to one responsibility per method.